gc.c

```diff
@@ -10879,7 +10915,7 @@ gc_verify_compaction_references(rb_execution_context_t *ec, VALUE self, VALUE do
         }
 
         if (RTEST(toward_empty)) {
-            gc_sort_heap_by_empty_slots(objspace);
+            objspace->rcompactor.compare_func = compare_free_slots;
```

Comment:
Do we ever unset the `compare_func`? Wouldn't it be `compare_free_slots` for all subsequent compaction runs?

Reply:
Great spot! Thank you. verified and fixed.